### PR TITLE
fix uninitialized variable

### PR DESCRIPTION
**Pull request by mosfet80**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/TechDraw/Gui/CommandExtensionPack.cpp [hunk 1]
```diff
@@ -1446,7 +1446,7 @@ void CmdTechDrawExtensionPositionSectionView::activated(int iMsg) {
             QObject::tr("Selection is empty"));
         return;
     }
-    float xPos, yPos;
+    float xPos=NAN, yPos= NAN;
```
Comment (chennes) [inline src/Mod/TechDraw/Gui/CommandExtensionPack.cpp:1449]: If we're going to initialize these to NaN, shouldn't we also check for it before assigning, below? It looks like there is indeed a branch where it's possible they aren't set, and I'm not sure it's a good idea to let this code deliver NaN as a result for the sectionView position.

**Comment by WandererFan** [comment]
setting to NaN will fix the "uninitialized" warning, but a better solution would be: 
```
 auto objFeat = selection[0].getObject();
    if (objFeat->isDerivedFrom(TechDraw::DrawViewSection::getClassTypeId())) {
        TechDraw::DrawViewSection* sectionView = dynamic_cast<TechDraw::DrawViewSection*>(objFeat);
        TechDraw::DrawViewPart* baseView = sectionView->getBaseDVP();
        double xPos = baseView->X.getValue();
        double yPos = baseView->Y.getValue();
        if (baseView->isDerivedFrom(TechDraw::DrawProjGroupItem::getClassTypeId())) {
            std::vector<App::DocumentObject*> parentViews = baseView->getInList();
            if (!parentViews.empty()) {
                TechDraw::DrawProjGroup* groupBase = dynamic_cast<TechDraw::DrawProjGroup*>(parentViews[0]);
                xPos = groupBase->X.getValue();
                yPos = groupBase->Y.getValue();
            }
        }
        std::string direction = sectionView->SectionDirection.getValueAsString();
        if ((direction == "Right") || (direction == "Left"))
            sectionView->Y.setValue(yPos);
        else if ((direction == "Up") || (direction == "Down"))
            sectionView->X.setValue(xPos);
    }
```

File: src/Mod/TechDraw/Gui/CommandExtensionPack.cpp (hunk 1 above)
Comment (wwmayer) [reply src/Mod/TechDraw/Gui/CommandExtensionPack.cpp:1449]: I also think it's better to initialize them with 0.0 instead.

**Comment by wwmayer** [comment]
90659c131c6

**Comment by wwmayer** [comment]
> setting to NaN will fix the "uninitialized" warning, but a better solution would be:

There are a few issues with the code block (besides the initialization):
* the dynamic_cast of objFeat is not needed here because it's already checked that it's a DrawViewSection
* the dynamic_cast of `parentViews[0]` is fine but then the returned value must be checked for null
* `baseView` could be null which is not checked. Thus initializing xPos/yPos this way won't work because it must be done in the if-block

**Pull request closed without merging**